src/grouping-and-mapping/clients/GroupsClient.ts

```diff
@@ -73,38 +78,52 @@ export class GroupsClient extends OperationsBase  implements IGroupsClient {
-    if(!this.topIsValid(top)) {
-      throw new RequiredError(
-        "top",
-        "Parameter top was outside of the valid range [1-1000]."
-      );
+  public async getGroups(accessToken: AccessToken, mappingId: string, preferReturn?: PreferReturn.Minimal, top?: number): Promise<GroupMinimalList>;
+  public async getGroups(accessToken: AccessToken, mappingId: string, preferReturn: PreferReturn.Representation, top?: number): Promise<GroupList>;
+  public async getGroups(accessToken: AccessToken, mappingId: string, preferReturn?: PreferReturn, top?: number): Promise<GroupMinimalList | GroupList> {
+    if (top !== undefined && !this.topIsValid(top)) {
+      throw new RequiredError("top", "Parameter top was outside of the valid range [1-1000].");
     }
 
     const url = this.constructUrl(mappingId, undefined, top);
-    const request = this.createRequest("GET", accessToken);
-    const response =  await this.fetchJSON<GroupList>(url, request);
-    return response;
+    const request = this.createRequest("GET", accessToken, undefined, preferReturn);
+
+    if (preferReturn === PreferReturn.Representation) {
+      return this.fetchJSON<GroupList>(url, request);
+    } else {
+      return this.fetchJSON<GroupMinimalList>(url, request);
+    }
   }
 
-  public getGroupsIterator(accessToken: AccessToken,  mappingId: string, top?: number ): EntityListIterator<Group> {
-    if(!this.topIsValid(top)) {
+  public getGroupsIterator(accessToken: AccessToken, mappingId: string, preferReturn?: PreferReturn.Minimal, top?: number): EntityListIterator<GroupMinimal>;
+  public getGroupsIterator(accessToken: AccessToken, mappingId: string, preferReturn: PreferReturn.Representation, top?: number): EntityListIterator<Group>;
+  public getGroupsIterator(accessToken: AccessToken, mappingId: string, preferReturn?: PreferReturn, top?: number): EntityListIterator<Group> | EntityListIterator<GroupMinimal> {
+    if (!this.topIsValid(top)) {
       throw new RequiredError(
         "top",
         "Parameter top was outside of the valid range [1-1000]."
       );
     }
 
     const url = this.constructUrl(mappingId, undefined, top);
-    const request = this.createRequest("GET", accessToken);
-    return new EntityListIteratorImpl(async () => getEntityCollectionPage<Group>( url, async (nextUrl: string): Promise<Collection<Group>> => {
-      const response = await this.fetchJSON<GroupList>(nextUrl, request);
+    const request = this.createRequest("GET", accessToken, undefined, preferReturn);
+
+    if (preferReturn === PreferReturn.Representation) {
+      return new EntityListIteratorImpl(async () => this.fetchCollection<Group>(url, request));
+    } else {
+      return new EntityListIteratorImpl(async () => this.fetchCollection<GroupMinimal>(url, request));
+    }
+  }
+
+  protected async fetchCollection<T extends GroupMinimal | Group>(url: string, request: RequestInit): Promise<EntityCollectionPage<T>> {
```

Comment:
Why protected?

Reply:
Maybe for future extensibility (not likely). But private makes more sense now.